moc_exceptions.py

```diff
@@ -0,0 +1,33 @@
+class InvalidEmailError(Exception):
+    """User's email address does not pass basic format validation"""
+
+
+class ItemExistsError(Exception):
+    """Keystone resource already exists and cannot be created"""
+    def __init__(self, item_type, item_name):
+        msg = "{} exists with the name: {}".format(item_type, item_name)
+        super(ItemExistsError, self).__init__(msg)
+
+
+class ItemNotFoundError(Exception):
+    """The specified Keystone resource was not found"""
+    def __init__(self, item_type, item_name):
+        msg = "No {} found in Keystone with name: {}".format(item_type,
+                                                             item_name)
+        super(ItemNotFoundError, self).__init__(msg)
+
+
+class BadEmailRecipient(Exception):
+    """If sending failed to one or more recipients, but not all of them."""
+    def __init__(self, rdict, subject):
+        self.__name__ = 'BadEmailRecipient'
+        self.rejected = rdict
+        self.message = ("Message '{0}' could not be sent to one or more "
+                        "recipients.").format(subject)
+
+
+class ProjectNotFoundError(Exception):
```

Comment:
A note for other reviewers - I suspect we can probably replace this with `ItemNotFoundError` but it is probably best left for a new PR, Duaa may not have time before she leaves at the end of the week.